codegen/src/main/twirl/templates/ScalaClient/Client.scala.txt

```diff
@@ -7,27 +7,37 @@
 @akka.grpc.gen.Constants.DoNotEditComment
 package @service.packageName
 
-import scala.concurrent.{ ExecutionContext, Promise }
+import scala.concurrent.ExecutionContext
 
 import io.grpc.ManagedChannel
 import io.grpc.MethodDescriptor
 
 import akka.grpc.GrpcClientSettings
 
 import akka.grpc.scaladsl.AkkaGrpcClient
-import akka.grpc.scaladsl.SingleResponseRequestBuilder
-import akka.grpc.scaladsl.StreamResponseRequestBuilder
 
-import akka.grpc.internal.ScalaUnaryRequestBuilder
-import akka.grpc.internal.ScalaBidirectionalStreamingRequestBuilder
-import akka.grpc.internal.ScalaClientStreamingRequestBuilder
-import akka.grpc.internal.ScalaServerStreamingRequestBuilder
 import akka.grpc.internal.Marshaller
 import akka.grpc.internal.NettyClientUtils
 import akka.grpc.internal.ClientState
 
 import akka.stream.Materializer
 
+@{
+  def withSingleResponse(stmt: String) = Set("import akka.grpc.scaladsl.SingleResponseRequestBuilder", stmt)
+  def withStreamResponse(stmt: String) = Set("import akka.grpc.scaladsl.StreamResponseRequestBuilder", stmt)
+  service.methods.toSet.flatMap { method: akka.grpc.gen.scaladsl.Method =>
```

Comment:
I have no idea why the type can't be inferred here actually.

```
[error] /Users/tz70s/WorkSpace/scala/akka-grpc/codegen/src/main/twirl/templates/ScalaClient/Client.scala.txt:28:35: missing parameter type
[error]   service.methods.toSet.flatMap { method =>
[error]                                   ^
[error] one error found
[error] (akka-grpc-codegen / Compile / compileIncremental) Compilation failed
```

Reply:
Probably because Set is contravariant. You could try

```suggestion
  service.methods.distinct.flatMap { method =>
```

Reply:
Thanks for suggestion! 

It's because `List` is covariant and `Set` is invariant (ref: https://stackoverflow.com/questions/26237939/toset-and-type-inference)

I think I need a `Set` here for flatMapping to avoid duplicate import statements. 
Another options are flatMapping `List` and toSet at the final step.
```scala
service.methods.flatMap { /* generate statements */ }.toSet.mkString("\n")
```
and create a temporary value for type inference.
```scala
val methodsSet = service.methods.toSet
methodsSet.flatMap { /* generate statements */ }.mkString("\n")
```

Anyway, I guess the explicit type signature is pretty safe here?